static/js/dimmable-light.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Dimmable Light
+ *
+ * UI element representing a light with an independent dimming function.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+'use strict';
+
+/* globals API, MultiLevelSwitch */
+
+/**
+ * DimmableLight Constructor (extends MultiLevelSwitch).
+ *
+ * @param Object description Thing description object.
+ * @param {String} format 'svg', 'html', or 'htmlDetail'.
+ */
+function DimmableLight(description, format) {
+  if (format === 'htmlDetail') {
+    this.details = this.details || {};
+    this.details.level = new LevelDetail(this);
```

Comment:
This is already taken care of by MultiLevelSwitch

Reply:
Good catch!